iDiff/utils/docker_utils.go

```diff
@@ -0,0 +1,172 @@
+package utils
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"os/exec"
+	"regexp"
+	"strings"
+	"syscall"
+
+	img "github.com/docker/docker/api/types/image"
+	"github.com/docker/docker/client"
+	"github.com/golang/glog"
+)
+
+// ValidDockerVersion determines if there is a Docker client of the necessary version locally installed.
+func ValidDockerVersion() (bool, error) {
+	cli, err := client.NewEnvClient()
+	if err != nil {
+		return false, fmt.Errorf("Docker client error: %s", err)
+	}
+	version := cli.ClientVersion()
+	if version == "1.31" {
```

Comment:
In attempting to use this branch I am getting: `Error response from daemon: client is newer than server (client 
API version: 1.31, server API version: 1.24)`.  The error is related to client and server incompatibility.  This check doesn't seem sufficient, is cli.ClientVersion output the version of the lib:
https://github.com/moby/moby/blob/master/client/client.go#L236
so I think this will always be true.  I think you would need to get the server version and make sure 1.31 is compatible with that as the check.  Perhaps you can add a flag to use onboard docker and output to use the flag in case of an error or make that the default for now?